pkg/querier/http.go

```diff
@@ -326,3 +337,17 @@ func writeError(err error, w http.ResponseWriter) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func (q *Querier) validateEntriesLimits(ctx context.Context, limit uint32) error {
```

Comment:
You need to validate this also for `LogQueryHandler` but only once we have verified it's a log request.

So after those lines:

```
	// short circuit metric queries
	if _, ok := expr.(logql.SampleExpr); ok {
		writeError(httpgrpc.Errorf(http.StatusBadRequest, "legacy endpoints only support %s result type", logql.ValueTypeStreams), w)
		return
	}
```